ASSIGNMENT ON MEASURES OF CENTRAL TENDENCY AND DISPERSION

In [ ]:
import numpy as np
import pandas as pd


In [ ]:
data=pd.read_csv('/content/StudentsPerformance.csv')

In [ ]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [46]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [ ]:
data.mean()

<ipython-input-3-abc01cf6c622>:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  data.mean()


math score       66.001
reading score    69.195
writing score    68.119
dtype: float64

1. FIND THE NUMBER OF MALES  AND FEMALES PARTICIPATED IN THE TEST

In [ ]:
gender = data["gender"].value_counts()
gender

female    518
male      482
Name: gender, dtype: int64

 2.About the Students' parental level of education

In [14]:
data['parental level of education'].value_counts()


some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Conclusion

226 parents have college education.
out of 226, 118 have bachelors degree and 59 have masters degree.

### 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course


In [36]:
import pandas as pd
data=pd.read_csv('/content/StudentsPerformance.csv')

In [37]:
 data1=data[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').mean()


In [38]:
data1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Conclusion:
 Female students have higher average in reading and writing score, male students have higher average in maths score.

In [40]:
import pandas as pd

In [39]:
data=pd.read_csv('/content/StudentsPerformance.csv')

In [41]:
data2= data[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').mean()
data2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Conclusion:
Students who have done test preparation course have higher average score in reading ,writing and math score.

4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course

In [42]:
data3= data[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').std()

In [43]:
data3

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

Conclusion: male students have less variations across each of the scores.

5. The management needs your help to give bonus points to the top 25% of
students based on their math score, so how will you help the management
to achieve this.

In [48]:
data.nlargest(250, ['math score'])

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]